### PR TITLE
fix: open datalog arguments

### LINKED ISSUE #5068: Add `OpenSchema` expression
Given the program:

```
def isCyclic(db: #{Edge(Int32, Int32)}): Bool = 
    let pr = #{
        Path(x, y) :- Edge(x, y).
        Path(x, z) :- Path(x, y), Edge(y, z).
    };
    not (query db, pr select () from Cycle() |> List.isEmpty)
```

We get the type error:

```
❌ -- Type Error --------------------------------------------------

>> Unable to unify the types: '#(  )' and '#( Path(Int32, Int32) | s0 )'.

6 |     not (query db, pr select () from Cycle() |> List.isEmpty)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
             mismatched types.

Type One: #{ Edge(Int32, Int32) }
Type Two: #{ Edge(t0, t0), Path(t0, t0) | s1 }
```

because the argument `db` is a ***closed row***.

The type of `db` is `#{Edge = ... | {}}`. Notice that this type (which is like a list) ends in `{}`. The idea is for the compiler to change this to `#{Edge = ... | r}` for a fresh type variable `r`. This needs to happen around here:

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Typer.scala#L305

where `getSubstFromParams` should open the rows.

Note that the transformation must ***only*** take place if there is a closed row.

### PR DESCRIPTION
**Pull request by JonathanStarup**
Fixes #5068

### CODE REVIEW

**Comment by JonathanStarup** [comment]
RFR @magnus-madsen

File: main/test/flix/Test.Exp.Fixpoint.Compose.flix [hunk 1]
```diff
@@ -86,4 +86,14 @@ namespace Test/Exp/Fixpoint/Compose {
         let rb = solve c3 <+> c4;
         (ra, rb)
 
+    @test
+    def testFixpointCompose13(): Bool = {
+        let ans = query openRow(#{Coolio("quack!").}) select s from Duck(s);
+        ans == List#{"quack!"}
+    }
+
+    def openRow(p: #{Coolio(String)}): #{Coolio(String), Duck(String) | r} = {
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.Fixpoint.Compose.flix:95]: I would like some more examples: Say: 

- 1x with 1 predicate
- 2x with 2 predicates (just trying different orders)
- 4x with 3 predicates (just trying different orders)

Comment (magnus-madsen) [reply main/test/flix/Test.Exp.Fixpoint.Compose.flix:95]: Something that checks the order is not important.

Comment (JonathanStarup) [reply main/test/flix/Test.Exp.Fixpoint.Compose.flix:95]: I'm not sure I follow, you want this test copied with 2 and 3 predicates?

**Pull request merged**